source/assets/scripts/recipe-add.js

```diff
@@ -114,12 +114,20 @@ function addNewRecipe() {
 
     console.log(recipe);
 
-    try {
-      backend.add_recipe(recipe, true);  // using the backend to simply logic
-      window.location.assign('index.html');
-    } catch(e) {
-      alert(e);
-    }
+    var file = document.querySelector('input[type=file]').files[0];
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.addEventListener("load",()=>{
+      localStorage.setItem(`!${recipe.servings}${recipe.name}${recipe.readyInMinutes}`,reader.result);
+      recipe.thumbnail=localStorage.getItem(`!${recipe.servings}${recipe.name}${recipe.readyInMinutes}`);
+      try {
+        backend.add_recipe(recipe, true);  // using the backend to simply logic
+        window.location.assign('index.html');
+      } catch(e) {
+        alert(e);
+      }
+    });
+    
```

Comment:
the code to fetch the data from the upload and store them in local storage which can be access by url